### PR TITLE
Seal internal types in runtime libs

### PR DESCRIPTION
**Pull request by NewellClark**
While working on [this analyzer](https://github.com/dotnet/runtime/issues/49944) ([PR](https://github.com/dotnet/roslyn-analyzers/pull/5594)) I ran the analyzer on dotnet/runtime and fixed violations.

### CODE REVIEW

**Review by danmoseley — APPROVED** [review]
thanks!

**Review by stephentoub — COMMENTED** [review]
Thanks for doing this.

File: src/libraries/System.Text.Json/src/System/Text/Json/JsonPropertyDictionary.ValueCollection.cs [hunk 1]
```diff
@@ -6,7 +6,7 @@
 
 namespace System.Text.Json
 {
-    internal partial class JsonPropertyDictionary<T>
+    internal sealed partial class JsonPropertyDictionary<T>
```
[the remark this answers is not in the document]
Comment (NewellClark) [reply src/libraries/System.Text.Json/src/System/Text/Json/JsonPropertyDictionary.ValueCollection.cs:9]: Good catch. I'll edit the analyzer so it reports all locations.

File: src/libraries/Common/src/System/Net/ContextAwareResult.cs [hunk 2]
```diff
@@ -68,7 +68,7 @@ internal bool IsCompatible(AsyncCallback? callback)
 
     // This class will ensure that the correct context is restored on the thread before invoking
     // a user callback.
-    internal partial class ContextAwareResult : LazyAsyncResult
+    internal sealed partial class ContextAwareResult : LazyAsyncResult
```
[the remark this answers is not in the document]
Comment (NewellClark) [reply src/libraries/Common/src/System/Net/ContextAwareResult.cs:71]: Fixed.

File: src/libraries/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/Semantics/MethodIterator.cs [hunk 3]
```diff
@@ -10,7 +10,7 @@ namespace Microsoft.CSharp.RuntimeBinder.Semantics
 {
     internal sealed partial class CMemberLookupResults
     {
-        public class CMethodIterator
+        public sealed class CMethodIterator
```
[the remark this answers is not in the document]
Comment (cston) [reply src/libraries/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/Semantics/MethodIterator.cs:13]: Adding @jaredpar.

[the remark this answers is not in the document]
Comment (cston) [reply src/libraries/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/Semantics/MethodIterator.cs:13]: > would you prefer we exempt Microsoft.CSharp from this analyzer via a NoWarn, or are you ok with such minimal changes?

The changes here (to seal internal types in Microsoft.CSharp and Microsoft.VisualBasic.Core) look good, thanks.

**Review by cston — APPROVED** [review]

**Pull request merged**